chore: Root all API URLs

johncowen opened the pull request:
We mostly hardcode all our URLs and hide them within a 'data retrieval' class. When I say mostly, we have complete hardcode-able control over them, but they are made up of dynamic segments. As we have complete control over them, we can hardcode every one of them to be 'rooted'. i.e. `/meshes/${name}` vs `meshes/${name}`

We also have almost complete control over the baseURL for all our API requests from our application. This means we can ensure that our baseURL does not have a trailing slash i.e. `http://localhost:5681` vs `http://localhost:5681/` (it's already like this in our development build, so no change for this in this PR)

This means that if we stick to the "rooted URLs with a non trailing slash baseURL" rule. In order to concatenate URLs we just do:

```
${baseURL}${path}
```

...which removes any complications, plus the need for any tests that go with testing any of that complication, and any code scanning issues in relation to the regexs we were using. 

Notes:
- We continue to use a prefix of `http` to decide whether something requires a `baseURL` adding or not.
- We 'almost' have complete control over our baseURL. During development we control this value via a hardcoded env var (which currently has no trailing slash), but in production it is passed to us via the backend. I'm currently unsure whether this is validated/processed to ensure it has no trailing slashes before being passed to us. Ideally I would slightly prefer it to be done in the backend, but if we can't do that for whatever reason, I can add a `stripLastSlashes(baseURL)` on the frontend instead (possibly a post-review follow up commit on this PR).
- There's probably a few more RestClient tests I could remove here as they aren't really testing anything apart from `${baseURL}${path}` which isn't really worth testing.

~Last note in case, before merging this a decision needs to be made on where we do the "remove any trailing slashes from the baseURL during production" i.e. **This PR should not be merged until this little decision is made.**~

We decided to keep the URL sanitizing/normalizing in the frontend, we've put it as close to where it comes in from the backend as possible.


Signed-off-by: John Cowen <john.cowen@konghq.com>

johncowen commented on src/services/kuma-api/KumaApi.ts:
```diff
@@ -200,18 +200,14 @@ export default class KumaApi extends Api {
-  }
-
-  getAllPolicyEntities({ path }: { path: string }, params?: PaginationParameters): Promise<PaginatedApiListResponse<PolicyEntity>> {
-    return this.client.get(path, { params })
```
I removed `getAllPolicyEntities` as we no longer use it anywhere.

lahabana reviewed (commented):
The change to KumaApi makes sense (it's customary to use absolute paths on clients).
However, as discussed offline I think not expecting anything from baseUrl (trailing or not trailing) will be more future proof (this comes from different places and it would make thing brittle. In the end it's simple enough to do:

```typescript
var baseUrl = ${url.startsWith('http') ? '' : this.baseUrl}
if (baseUrl.endsWith('/')) { // Strip last '/'
  baseUrl = baseUrl.substring(0, baseUrl.length - 1);
}
return makeRequest(${baseUrl}${url}, normalizedOptions,payload);
```

johncowen commented:
Ok there was already some normalizing/sanitizing happening when we pick the API baseUrl out of the JSON in the HTML. I changed it around a bit to use less/no regex, which was the original issue we were tackling here. See https://github.com/kumahq/kuma-gui/pull/785/commits/0cfbb53b3b8d0900b916b2bb25ec296e22e7bc23.

I also removed being able to set `KUMA_API_URL` directly (we should always pull it from the HTML), and changed the tests to be more unit test rather than integration tests (I believe being able to set `KUMA_API_URL` directly was only added to enable the style of testing used previous to this PR)

johncowen commented on src/services/env/Env.ts:
```diff
@@ -63,23 +62,35 @@ export default class Env {
       }
     }
+    if (!config) {
+      config = getPathConfigDefault(import.meta.env.PROD ? '/' : import.meta.env.VITE_KUMA_API_SERVER_URL)
```
😆 thanks! great spot!
